src/utility_linux.go

```diff
@@ -24,7 +24,23 @@ func decodeString(fileContent string) (string, error) {
 
 // sendNotification sends a notification to the end user.
 func sendNotification(mc *metaConfig, messageString string) {
-	shellCommand(`l_display=":$(ls /tmp/.X11-unix/* | sed 's#/tmp/.X11-unix/X##' | head -n 1)"; l_user=$(who | grep '('$display')' | awk '{print $1}' | head -n 1); if [ -z "$l_user" ]; then l_user="` + mc.Config.User + `"; fi; l_uid=$(id -u $l_user); sudo -u $l_user DISPLAY=$l_display DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/$l_uid/bus notify-send -i /opt/aeacus/assets/logo.png "Aeacus SE" "` + messageString + `"`)
+	if mc.Config.User == "" {
+		failPrint("User not specified in configuration, can't send notification.")
+	} else {
+		shellCommand(`
+			user="` + mc.Config.User + `"
+			uid="$(id -u $user)"
+
+			# If bus exists, Ubuntu >= 18
+			if [ -e /run/user/$uid/bus ]; then
+			    display="unix:path=/run/user/$uid/bus"
+			else
+			# Ubuntu <= 16
+			    display="unix:abstract=$(cat /run/user/$uid/dbus-session | cut -d '=' -f3)"
+			fi
+
+			sudo -u $user DISPLAY=:0 DBUS_SESSION_BUS_ADDRESS=$display notify-send -i /opt/aeacus/assets/logo.png "Aeacus SE" "` + messageString + `"`)
+	}
```

Comment:
What if someone did put in the user in scoring.conf and they are on ubuntu 16

Reply:
That's what it's checking for

Reply:
`/run/user/$uid/bus` is a symlink that's only in the newer versions of Ubuntu, `-e` returns true if any file (of any type) exists at the specified path.